### ISSUE TITLE
CAOs vanish for one frame when shifting camera offset

### ISSUE THREAD

**Issue by hecktest** [minetest/minetest#10027]
_5.3.0-dev_
This happens when the camera offset changes, every 200 nodes traveled. Every CAO including the player vanishes for a frame.
![blink](https://user-images.githubusercontent.com/42101236/84343409-6674af80-aba8-11ea-8b84-2fffa3931ff2.gif)
![framestep](https://user-images.githubusercontent.com/42101236/84343302-10077100-aba8-11ea-95bf-cc5eceb4eed0.gif)
I tried looking for a quick and dirty fix but it's not as easy as shifting some function calls around. Any help?

**Comment by sfan5** [comment]
confirmed on 80d9e9c1834fdc78588ffe2842766c0177227786

When changing `CAMERA_OFFSET_STEP` to 20 it becomes apparent that the player doesn't in fact disappear, but is rendered offset by 20 nodes:
![mpv-shot0001](https://user-images.githubusercontent.com/1042418/84368943-7c00ce00-abd6-11ea-860a-19f96972d44e.jpg)

**Comment by oilboi** [comment]
Oh no :⭕

**Comment by hecktest** [comment]
Yes, I suspected as much. The CAOs are displayed with a stale position, because they are updated before the camera. But updating the camera before step causes it to lag behind by a frame.
https://github.com/minetest/minetest/blob/80d9e9c1834fdc78588ffe2842766c0177227786/src/client/game.cpp#L1147

**Comment by erlehmann** [comment]
I can confirm that this still happens as of commit 1b2176a426dd987795f20e9042a8d79f958b7b44.

**Comment by lhofhansl** [comment]
Has anyone found a solution. As a quick hack to reduce we can increase CAMERA_OFFSET to 1000. Since this only guards against rounding errors in the shaders (and other client side rendering?) 1000 should be good and reduce the problem 5x.

Not a fix... Obviously.

**Comment by IrrlichtMt4** [comment]
I noticed not only does the player disappear in the video but also all the entities. Notice how the orange stuff also disappear in the exact frames the player flickers.

**Comment by appgurueu** [comment]
Yay! Glad to see this finally closed.
